### PR TITLE
Removes proto and golang tooling

### LINKED ISSUE #102: Update to recent protoc-gen-go deps
getenvoy can't update to recent dependencies because the generated code here is still using old protobuf

please migrate from github.com/golang/protobuf to google.golang.org/protobuf

Note: you can probably use buf.build instead of protoc as it can be managed with tools like bingo

### PR DESCRIPTION
**Pull request by codefromthecrypt**
manifest.proto moves to the canonical repo https://github.com/tetratelabs/getenvoy

Fixes #102

### CODE REVIEW

File: .bingo/Variables.mk [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	@# Install binary/ries using Go 1.14+ build command. This is using bwplotka/bingo-controlled, separate go module with pinned dependencies.
+	@echo "(re)installing $(GOBIN)/licenser-v0.6.0"
+	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=licenser.mod -o=$(GOBIN)/licenser-v0.6.0 "github.com/liamawhite/licenser"
+
+PROTOC_GEN_GO := $(GOBIN)/protoc-gen-go-v1.26.0
+$(PROTOC_GEN_GO): $(BINGO_DIR)/protoc-gen-go.mod
+	@# Install binary/ries using Go 1.14+ build command. This is using bwplotka/bingo-controlled, separate go module with pinned dependencies.
+	@echo "(re)installing $(GOBIN)/protoc-gen-go-v1.26.0"
+	@cd $(BINGO_DIR) && $(GO) build -mod=mod -modfile=protoc-gen-go.mod -o=$(GOBIN)/protoc-gen-go-v1.26.0 "google.golang.org/protobuf/cmd/protoc-gen-go"
+
```
Comment (codefromthecrypt) [inline .bingo/Variables.mk:49]:
```suggestion
```

File: Makefile [hunk 2]
```diff
@@ -47,10 +50,70 @@ push: $(TARGETS:=.push)
...
+	    $(GOIMPORTS) -w -local github.com/tetratelabs/getenvoy-package $$f; \
+	done
+
+# Enforce go version matches what's in go.mod when running `make check` assuming the following:
+# * 'go version' returns output like "go version go1.16 darwin/amd64"
+# * go.mod contains a line like "go 1.16"
+EXPECTED_GO_VERSION_PREFIX := "go version go$(shell sed -ne '/^go /s/.* //gp' go.mod )"
+GO_VERSION := $(shell go version)
+
+.PHONY: check
```
Comment (codefromthecrypt) [inline Makefile:96]: this is the new pre-flight check which is similar to what's happening in https://github.com/tetratelabs/getenvoy except this has a protobuf (api) step

File: go.mod [hunk 3]
```diff
@@ -1,5 +1,5 @@
 module github.com/tetratelabs/getenvoy-package
 
-go 1.12
+go 1.16
 
-require github.com/golang/protobuf v1.3.2
+require google.golang.org/protobuf v1.26.0
```
Comment (codefromthecrypt) [inline go.mod:5]: this is the important part for downstream

File: Makefile [hunk 4]
```diff
@@ -47,10 +50,70 @@ push: $(TARGETS:=.push)
 release: $(TARGETS:=.release)
 	@true
 
-clean:
-	rm -rf $(OUTPUT_DIR)
-
 license:
 	licenser apply -r . "Tetrate"
 
+
```
Comment (codefromthecrypt) [inline Makefile:56]:
```suggestion
```

File: envoy_pkg/package_envoy.py [hunk 5]
```diff
@@ -16,6 +16,7 @@
 
 import os
 import sys
+
```
Comment (codefromthecrypt) [inline envoy_pkg/package_envoy.py:19]: odd this was only caught now..

File: api/manifest.proto [hunk 6]
```diff
@@ -15,6 +15,8 @@
 syntax = "proto3";
 package api;
 
+option go_package = "github.com/tetratelabs/getenvoy-package/api;api";
```
Comment (codefromthecrypt) [inline api/manifest.proto:18]: @lizan can we move this proto and the things to generate it to tetratelabs/getenvoy? As far as I can tell, the only non-deprecated consumer of this manifest besides getenvoy is https://github.com/tetratelabs/getenvoy-ci/blob/master/go.mod I don't see the sense in remote coupling to a protobuf version over that, as ci should be able to import getenvoy directly if it needed to, right?

Comment (mathetake) [reply api/manifest.proto:18]: actually it is used in tetratelabs/getenvoy-ci's manifest/validate_manifest_test.go to validate manifest/manifest.json.. but yeah we may want reconsider where to put these stuff (including manifest.json and this proto)..

Comment (lizan) [reply api/manifest.proto:18]: Yeah we should move this there, no objection from me.

Comment (codefromthecrypt) [reply api/manifest.proto:18]: I'll own migrating getenvoy-ci following this. thanks for the feedback!

**Comment by codefromthecrypt** [comment]
I'll repurpose this PR to delete the proto stuff instead of adding more tooling

**Comment by codefromthecrypt** [comment]
the change to getenvoy is in, but I've not reworked this yet to delete the proto. I will

**Comment by codefromthecrypt** [comment]
ok this is updated

**Review by mathetake — APPROVED** [review]

**Comment by codefromthecrypt** [comment]
https://github.com/tetratelabs/getenvoy-ci/pull/233 for cleanup

**Pull request merged**